### PR TITLE
frame: rework flags

### PR DESCRIPTION
**Pull request by dshil**
Relates: #115

### CODE REVIEW

File: src/tests/roc_audio/test_depacketizer.cpp [hunk 1]
```diff
@@ -300,74 +301,136 @@ TEST(depacketizer, overlapping_packets) {
...
+        new_packet(SamplesPerPacket * 6, 0.11f),
+        new_packet(SamplesPerPacket * 2, 0.11f),
+        new_packet(SamplesPerPacket * 3, 0.11f),
+    };
 
-    queue.write(new_packet(ts1, 0.11f));
-    queue.write(new_packet(ts2, 0.22f));
-    queue.write(new_packet(ts3, 0.33f));
+    unsigned frame_flags[] = {
+        Frame::FlagFull | !(Frame::FlagPacketDrops),
```
Comment (gavv) [inline src/tests/roc_audio/test_depacketizer.cpp]: `!` is a logical operation, not a bitwise one. It returns either 0 or 1.

File: src/tests/roc_audio/test_depacketizer.cpp [hunk 2]
```diff
@@ -300,74 +301,136 @@ TEST(depacketizer, overlapping_packets) {
...
+    for (size_t n = 0; n < ROC_ARRAY_SIZE(packets); n++) {
+        queue.write(packets[n]);
+    }
+
+    for (size_t n = 0; n < ROC_ARRAY_SIZE(packets); n++) {
+        Frame frame = new_frame(SamplesPerPacket);
+        dp.read(frame);
 
-    CHECK(frame.flags() & Frame::FlagSkip);
+        CHECK(frame.flags() & frame_flags[n]);
```
Comment (gavv) [inline src/tests/roc_audio/test_depacketizer.cpp]: Why do we need `&` here? Why not just listing the exact expected flags in `frame_flags` and then checking them here?

```
CHECK(frame.flags() == frame_flags[n]);
```

File: src/tests/roc_audio/test_depacketizer.cpp [hunk 3]
```diff
@@ -300,74 +301,136 @@ TEST(depacketizer, overlapping_packets) {
...
+        if (packets[n] != NULL) {
+            queue.write(packets[n]);
+        }
 
-    CHECK(!(frame.flags() & Frame::FlagSkip));
+        Frame frame = new_frame(frame_sizes[n]);
+        dp.read(frame);
+
+        if (frame_flags[n] == Frame::FlagFull) {
+            CHECK(frame.flags() & Frame::FlagFull);
```
Comment (gavv) [inline src/tests/roc_audio/test_depacketizer.cpp]: Same note here, why not just listing and checking the exact flags?

File: src/tests/roc_audio/test_depacketizer.cpp (hunk 1 above)
Comment (gavv) [reply src/tests/roc_audio/test_depacketizer.cpp]: BTW, GCC 7 even generates a warning (see failed build):

```
src/tests/roc_audio/test_depacketizer.cpp:318:51: error: 
   enum constant in boolean context [-Werror=int-in-bool-context]
         Frame::FlagFull | !(Frame::FlagPacketDrops),
```

File: src/tests/roc_audio/test_depacketizer.cpp (hunk 3 above)
Comment (dshil) [reply src/tests/roc_audio/test_depacketizer.cpp]: Yep, I've fixed it.

File: src/tests/roc_audio/test_depacketizer.cpp (hunk 2 above)
Comment (dshil) [reply src/tests/roc_audio/test_depacketizer.cpp]: I've fixed it.

File: src/tests/roc_audio/test_depacketizer.cpp (hunk 1 above)
Comment (dshil) [reply src/tests/roc_audio/test_depacketizer.cpp]: Damn..it's awful, I've fixed it.

**Pull request merged**